tools/sort.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+# Shell script equivalent of Metamath SORT tool
+# Mario Carneiro 7-Jan-2022
+
+usage() {
+  cat << HELP
+SORT - Sort the lines in a file with key starting at specified string
+
+This command sorts a file, comparing lines starting at a key string.
+If the key string is blank, the line is compared starting at column 1.
+If a line doesn't contain the key, it is compared starting at column 1.
+Syntax:  sort.sh KEY < INFILE > OUTFILE
+HELP
```

Comment:
I'd rather not ackowledge it in the docs, it's not part of the spec. The right solution is to implement this in python or other regular programming language as I mentioned, this is just a limitation of the quick and dirty solution that hopefully will not interfere with standard text manipulation activities.

Reply:
..hopefully...  That is precisely what my comment is about.  Mark it as an implementation detail, workaround or todo, but mention it.  It describes the behaviour.  For the overall concept see here https://stackoverflow.com/questions/35303332/what-are-preconditions-and-postconditions

Reply:
Well, this is why the original version was throwing an error, because I didn't want to deal with this. It's tricky to throw an error in the middle of an awk script, but I put something that should work well enough.